rpc/block_svc_routines.c

```diff
@@ -2478,6 +2485,333 @@ block_replace_cli_1_svc_st(blockReplaceCli *blk, struct svc_req *rqstp)
+    }
+  }
+  // },
+  // ]
+
+  json_object_object_add(tg_obj, "tpgs", tpgs_arr);
+
+  snprintf(iqn, 128, "iqn.2016-12.org.gluster-block:%s", info->gbid);
+  json_object_object_add(tg_obj, "wwn", GB_JSON_OBJ_TO_STR(iqn));
+
+  return tg_obj;
+}
+
+
+struct json_object *
+getSoObj(char *block, MetaInfo *info, blockGenConfigCli *blk)
+{
+  char cfgstr[1024] = {'\0', };
+  struct json_object *so_obj = json_object_new_object();
+  struct json_object *so_obj_attr = json_object_new_object();
+
+
+  json_object_object_add(so_obj_attr, "cmd_time_out", json_object_new_int(0));
+  json_object_object_add(so_obj_attr, "dev_size", json_object_new_int64(info->size));
+
+  json_object_object_add(so_obj, "attributes", so_obj_attr);
+
+  snprintf(cfgstr, 1024, "glfs/%s@%s/block-store/%s", info->volume, blk->addr, info->gbid);
+  json_object_object_add(so_obj, "config", GB_JSON_OBJ_TO_STR(cfgstr));
+  json_object_object_add(so_obj, "name", GB_JSON_OBJ_TO_STR(block));
+  json_object_object_add(so_obj, "plugin", GB_JSON_OBJ_TO_STR("user"));
+  json_object_object_add(so_obj, "size", json_object_new_int64(info->size));
+  json_object_object_add(so_obj, "wwn", GB_JSON_OBJ_TO_STR(info->gbid));
+
+  return so_obj;
+}
+
+
+static int
+getSoTgArraysForAllVolume(struct soTgObj *obj, blockGenConfigCli *blk,
+                          char **errMsg, int *errCode)
+{
+  struct glfs *glfs;
+  struct glfs_fd *lkfd = NULL;
+  struct glfs_fd *tgmdfd = NULL;
+  struct dirent *entry;
+  MetaInfo *info = NULL;
+  strToCharArrayDefPtr vols;
+  size_t i, j;
+  int ret = -1;
+  bool partOfBlock;
+  struct json_object *so_obj = NULL;
+  struct json_object *tg_obj = NULL;
+
+
+  vols = getCharArrayFromDelimitedStr(blk->volume, GB_VOLS_DELIMITER);
+  if (!vols) {
+    LOG("mgmt", GB_LOG_ERROR,
+        "getCharArrayFromDelimitedStr(%s) failed", blk->volume);
+    goto optfail;
+  }
+
+  for (i = 0; i < vols->len; i++) {
+    glfs = glusterBlockVolumeInit(vols->data[i], errCode, errMsg);
+    if (!glfs) {
+      LOG("mgmt", GB_LOG_ERROR,
+          "glusterBlockVolumeInit(%s) failed", vols->data[i]);
+      goto optfail;
+    }
+
+    lkfd = glusterBlockCreateMetaLockFile(glfs, vols->data[i], errCode, errMsg);
+    if (!lkfd) {
+      LOG("mgmt", GB_LOG_ERROR, "%s %s", FAILED_CREATING_META, vols->data[i]);
+      goto optfail;
+    }
+
+    GB_METALOCK_OR_GOTO(lkfd, vols->data[i], *errCode, *errMsg, out);
+
+    tgmdfd = glfs_opendir(glfs, GB_METADIR);
+    if (!tgmdfd) {
+      *errCode = errno;
+      GB_ASPRINTF(errMsg, "Not able to open metadata directory for volume "
+          "%s[%s]", vols->data[i], strerror(*errCode));
+      LOG("mgmt", GB_LOG_ERROR, "glfs_opendir(%s): on volume %s failed[%s]",
+          GB_METADIR, vols->data[i], strerror(errno));
+      ret = -1;
+      goto out;
+    }
+
+    while ((entry = glfs_readdir(tgmdfd))) {
+      if (strcmp(entry->d_name, ".") &&
+          strcmp(entry->d_name, "..") &&
+          strcmp(entry->d_name, "meta.lock")) {
```

Comment:
You also need to add prio file here? Or make it a function and use it in all places to prevent missing it in some places